handler/src/main/java/io/netty/handler/logging/LoggingHandler.java

```diff
@@ -347,13 +348,13 @@ private static String formatByteBufHolder(ChannelHandlerContext ctx, String even
         ByteBuf content = msg.content();
         int length = content.readableBytes();
         if (length == 0) {
-            StringBuilder buf = new StringBuilder(chStr.length() + 1 + eventName.length() + 2 + msgStr.length() + 4);
+            StringBuilder buf = new StringBuilder(chStr.length() + eventName.length() + msgStr.length() + (1 + 2 + 4));
             buf.append(chStr).append(' ').append(eventName).append(", ").append(msgStr).append(", 0B");
             return buf.toString();
         } else {
             int rows = length / 16 + (length % 15 == 0? 0 : 1) + 4;
             StringBuilder buf = new StringBuilder(
-                    chStr.length() + 1 + eventName.length() + 2 + msgStr.length() + 2 + 10 + 1 + 2 + rows * 80);
+                    chStr.length() + eventName.length() + msgStr.length() + (1 + 2 + 2 + 10 + 1 + 2) + rows * 80);
```

Comment:
Moving these constants away from the things they're modifying makes it slightly less clear what's going on (acknowledging that's already true for `2 + 10 + 1 + 2`). I wonder if, given the number of small constants, it would make more sense to just smoosh everything down into a single constant (18), but add a comment explaining where it came from.

Reply:
I agree with @jchambers, yet I feel even more strongly about the loss of clarity.

I think it should be left as-is. I doubt this code is a performance bottleneck for anything (since it's inside a debugging handler), and I doubt these calculations are even a performance bottleneck for this handler in general. Rearranging these constants or collapsing leads to less clear code, and is the epitome of micro-optimizing at the expense of code quality. I know this quote is misused and misunderstood often, but here I think it's apt:

>Programmers waste enormous amounts of time thinking about, or worrying about, the speed of noncritical parts of their programs, and these attempts at efficiency actually have a strong negative impact when debugging and maintenance are considered.

Sorry @fenik17, I think you do wonderful work on the Netty codebase and I enjoy reading your PRs, but this particular change in the logging handler I think would be a mistake.

Reply:
@fenik17 I think I agree with @jchambers @jasontedor here ... clear code is more important here. So either I would do what @jchambers suggested or revert this part.

Reply:
I agree. This is a bit ugly code. Reverted..
I know how to do better, and maybe come back here later :)